src/vue/pages/withdrawals/WithdrawalRequestDetails.vue

```diff
@@ -0,0 +1,313 @@
+            <td>
+              {{ 'withdrawal-request-details.created' | globalize }}
+            </td>
+            <td>
+              {{ request.createdAt | formatDate }}
+            </td>
+          </tr>
+        </tbody>
+      </table>
+    </div>
+
+    <div class="withdrawal-request-details__buttons">
+      <button
+        v-ripple
+        class="withdrawal-request-details__update-btn"
+        :disabled="isPending || !canBeApproved || isShowReasonField"
+        @click="$emit(EVENTS.approve)"
+      >
+        {{ 'withdrawal-request-details.approve-btn' | globalize }}
+      </button>
+
+      <button
+        v-ripple
+        class="withdrawal-request-details__cancel-btn"
+        :disabled="isPending || !canBeRejected || isShowReasonField"
+        @click="showReasonField"
+      >
+        {{ 'withdrawal-request-details.rejected-btn' | globalize }}
+      </button>
+    </div>
+
+    <div
+      class="withdrawal-request-details__form-row"
+      v-if="isShowReasonField && !request.isPermanentlyRejected"
+    >
+      <div class="withdrawal-request-details__form-field">
+        <textarea-field
+          id="withdrawal-request-details-reject-reason"
+          name="withdrawal-request-details-reject-reason"
+          v-model="rejectReason"
+          :label="'withdrawal-request-details.reject-reason' | globalize"
+          :maxlength="formNoteMaxLength"
+        />
+      </div>
+    </div>
+
+    <div class="withdrawal-request-details__form-actions">
+      <form-confirmation
+        v-if="isShowReasonField && !request.isPermanentlyRejected"
+        :is-pending="isPending"
+        :message-id="'withdrawal-request-details.message-text-default'"
+        @cancel="showReasonField"
+        @ok="$emit(EVENTS.reject, rejectReason)"
+      />
+    </div>
+  </div>
+</template>
+
+<script>
+import { ASSET_POLICIES, REQUEST_TYPES } from '@tokend/js-sdk'
+
+import { REQUEST_STATES } from '@/js/const/request-states.const'
+
+import config from '@/config'
+
+import IdentityGetterMixin from '@/vue/mixins/identity-getter'
+
+import TextareaField from '@/vue/fields/TextareaField'
+import FormConfirmation from '@/vue/common/FormConfirmation'
+import {
+  WithdrawalDetailsRequestRecord,
+} from '@/js/records/requests/withdrawal-details.record'
+import EmailGetter from '@/vue/common/EmailGetter'
+
+const EVENTS = {
+  approve: 'approve',
+  reject: 'reject',
+}
+
+export default {
+  name: 'withdrawal-request-details',
+  components: {
+    TextareaField,
+    FormConfirmation,
+    EmailGetter,
+  },
+  mixins: [IdentityGetterMixin],
+  props: {
+    request: {
+      type: [WithdrawalDetailsRequestRecord],
+      required: true,
+    },
+    isPending: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+  },
+  data: _ => ({
+    rejectReason: '',
+    formNoteMaxLength: 250,
+    isShowReasonField: false,
+    config,
+    ASSET_POLICIES,
+    EVENTS,
+    REQUEST_STATES,
+    REQUEST_TYPES,
```

Comment:
Extra

Reply:
Recheck your PR for extra defines (include CSS classes)

Reply:
resolved